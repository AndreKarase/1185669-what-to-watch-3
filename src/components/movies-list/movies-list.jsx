import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import MovieCard from '../movie-card/movie-card.jsx';

class MoviesList extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {movie: null};
  }

  render() {
    const {movies, onHeaderClick} = this.props;

    return (
      <div className="catalog__movies-list">
        {movies.map((movie, i) => (<MovieCard
          key = {movie.title}
          movie = {movie}
          onHeaderClick = {() => {
            onHeaderClick(i);
          }}
          onMouseEnter = {(hoverMovie) => {
            this.setState({movie: hoverMovie});
          }}
          onMouseLeave = {() => {
            this.setState({movie: null});
          }}
        />))}
      </div>
    );
  }
}

MoviesList.propTypes = {
  movies: PropTypes.array.isRequired,
  onHeaderClick: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  movies: state.movies
});

export {MoviesList};
export default connect(mapStateToProps)(MoviesList);
