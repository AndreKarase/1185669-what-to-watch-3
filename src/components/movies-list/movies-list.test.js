import React from 'react';
import renderer from 'react-test-renderer';
import {MoviesList} from './movies-list';

const movies = [
  {
    title: `The Grand Budapest Hotel`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: ``
  }, {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: ``
  }, {
    title: `Bohemian Rhapsody`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: ``
  }
];

it(`Should render MoviesList`, () => {
  const tree = renderer.create(
      <MoviesList
        movies={movies}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
