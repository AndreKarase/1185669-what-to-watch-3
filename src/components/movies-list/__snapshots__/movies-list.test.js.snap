// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MoviesList 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="The Grand Budapest Hotel"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </div>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
        >
          Fantastic Beasts: The Crimes of Grindelwald
        </a>
      </h3>
    </div>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bohemian Rhapsody"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </div>
  </article>
</div>
`;
